sdk/client/src/broadcast_client.rs

```diff
@@ -0,0 +1,291 @@
+        start_seq: u64,
+        limit: u64,
+    ) -> Result<Response<Vec<EventView>>> {
+        let futures = self
+            .random_clients()
+            .map(|client| client.get_events(key, start_seq, limit));
+        let results = join_all(futures).await;
+        collect_results(results)
+    }
+
+    pub async fn get_currencies(&self) -> Result<Response<Vec<CurrencyInfoView>>> {
+        let futures = self.random_clients().map(|client| client.get_currencies());
+        let results = join_all(futures).await;
+        collect_results(results)
+    }
+
+    pub async fn get_network_status(&self) -> Result<Response<u64>> {
+        let futures = self
+            .random_clients()
+            .map(|client| client.get_network_status());
+        let results = join_all(futures).await;
+        collect_results(results)
+    }
+
+    //
+    // Experimental APIs
+    //
+
+    pub async fn get_state_proof(&self, from_version: u64) -> Result<Response<StateProofView>> {
+        let futures = self
+            .random_clients()
+            .map(|client| client.get_state_proof(from_version));
+        let results = join_all(futures).await;
+        collect_results(results)
+    }
+
+    pub async fn get_account_state_with_proof(
+        &self,
+        address: AccountAddress,
+        from_version: Option<u64>,
+        to_version: Option<u64>,
+    ) -> Result<Response<AccountStateWithProofView>> {
+        let futures = self
+            .random_clients()
+            .map(|client| client.get_account_state_with_proof(address, from_version, to_version));
+        let results = join_all(futures).await;
+        collect_results(results)
+    }
+
+    pub async fn get_transactions_with_proofs(
+        &self,
+        start_version: u64,
+        limit: u64,
+    ) -> Result<Response<()>> {
+        let futures = self
+            .random_clients()
+            .map(|client| client.get_transactions_with_proofs(start_version, limit));
+        let results = join_all(futures).await;
+        collect_results(results)
+    }
+
+    pub async fn get_events_with_proofs(
+        &self,
+        key: &str,
+        start_seq: u64,
+        limit: u64,
+    ) -> Result<Response<()>> {
+        let futures = self
+            .random_clients()
+            .map(|client| client.get_events_with_proofs(key, start_seq, limit));
+        let results = join_all(futures).await;
+        collect_results(results)
+    }
+}
+
+fn collect_results<T>(mut results: Vec<Result<T>>) -> Result<T> {
+    if results.is_empty() {
+        return Err(Error::unknown("Expected at least one response, found none"));
+    }
+    // All of them returned an error, so return the first error
+    if results.iter().all(|x| x.is_err()) {
+        if let Err(e) = results.swap_remove(0) {
+            return Err(e);
+        }
+    }
+    if let Some(result) = results.into_iter().filter_map(|x| x.ok()).nth(0) {
+        return Ok(result);
+    }
+    unreachable!()
```

Comment:
You should avoid using `unreachable` and prefer to restructure the code. Also take a look at https://github.com/diem/diem/blob/main/client/json-rpc/src/async_client/http_client.rs#L371 because we shouldn't just pick the first ok response, but rather the most up to date one.